packages/react-button/src/components/Button/ButtonBase.tsx

```diff
@@ -2,14 +2,15 @@ import * as React from 'react';
 import { ButtonProps } from './Button.types';
 import { compose, mergeProps } from '@fluentui/react-compose';
 import { useButton } from './useButton';
+import { useMergedRefs } from '@uifabric/react-hooks';
 
 export const ButtonBase = compose<'button', ButtonProps, ButtonProps, {}, {}>(
   (props, ref, options) => {
     const { state } = options;
     const { slots, slotProps } = mergeProps(state, options);
 
     return (
-      <slots.root ref={ref} {...slotProps.root}>
+      <slots.root ref={useMergedRefs(ref, state.buttonRef)} {...slotProps.root}>
```

Comment:
Shouldn't this merging of the refs be done in `useButton` instead?

Reply:
let me think about this.

Reply:
Another q: should ref even be here, and should it have been in slotProps.root already

Reply:
I think this is probably correct for now as the ref is being set here already. Let's keep as is for now.